daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/ContractStateMachine.scala

```diff
@@ -479,12 +508,12 @@ object ContractStateMachine {
     *        2.1. A create will set the corresponding map entry to KeyActive(cid) if the contract has a key.
     *        2.2. A consuming choice on cid will set the corresponding map entry to KeyInactive
```

Comment:
```suggestion
    *        2.2. A consuming exerciseByKey on cid will set the corresponding map entry to KeyInactive
```

Reply:
No, I really mean `exercise` here (renaming from choice though). If they key was brought in scope before (that’s what the rest of the sentence says) then an exercise drops it from the map even if that exercise was not by key.